app/components/work_package_types/projects_component.rb

```diff
@@ -46,16 +46,20 @@ def form_options
       }
     end
 
-    def projects = options[:projects]
-
     def build_project_tree(tree)
       nested_project_list = Project.build_projects_hierarchy(projects)
 
       add_sub_tree(tree, nested_project_list)
     end
 
+    def enabled_for_all_projects?
+      model.projects.pluck(:id).sort == projects.pluck(:id).sort
```

Comment:
I guess you are already aware of this, but we effectively still store the enabled state on each individual project and there is no global "enabled for all projects" setting.

I think UX-wise this will be confusing. In other places where the user can select a checkbox or toggle "enabled for all projects" it means "once and for all", so when a new project is added the user doesn't have to go in and enable it for this project again.

As far as I can tell, this is different here, because we do not store the "enabled for all" state anywhere. So once I create a new project, it will not be included and visiting the corresponding page will now show me the project selector again with a few projects (the new ones) unchecked.

Reply:
Mh. Okay, I think that's a topic for the bigger UI refactor in the end.

I wasn't aware of the "enabled for new projects" checkbox and I believe it belongs into this projects tabs. I'd even be inclined to auto-activate it when toggling "enabled for all projects" and otherwise showing it next to the tree view, because that would make sense. Then you either say "I want all projects ever" or "I want control over existing and new projects".

Reply:
I agree to you, but also to the point that it would require UX to rethink it first